### PR TITLE
added more emails and changed copy in old ones

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: app/models/form_answer.rb [hunk 1]
```diff
@@ -106,10 +106,12 @@ def secondary
   begin :scopes
     scope :for_award_type, -> (award_type) { where(award_type: award_type) }
     scope :for_year, -> (year) { joins(:award_year).where(award_years: { year: year }) }
-    scope :shortlisted_with_no_certificate, -> { where("1 = 0") }
-    scope :winners, -> { where("1 = 0") }
-    scope :unsuccessful, -> { where(state: %w(not_recommended reserved)) }
+    scope :shortlisted, -> { where(state: %w(reserved recommended)) }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 2]
```diff
@@ -106,10 +106,12 @@ def secondary
...
+    scope :not_shortlisted, -> { where(state: "not_recommended") }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 3]
```diff
@@ -106,10 +106,12 @@ def secondary
...
+    scope :winners, -> { where(state: %(recomended awarded)) }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 4]
```diff
@@ -106,10 +106,12 @@ def secondary
...
+    scope :unsuccessful, -> { where(state: %w(not_recommended not_awarded reserved)) }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 5]
```diff
@@ -106,10 +106,12 @@ def secondary
...
     scope :submitted, -> { where(submitted: true) }
+    scope :business, -> { where(award_type: %w(trade innovation development)) }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/services/notifiers/email_notification_service.rb [hunk 6]
```diff
@@ -18,34 +18,73 @@ def run
     end
   end
 
-  # this will be removed after all methods are implemented
-  %w(reminder_to_submit ep_reminder_support_letters winners_reminder_to_submit unsuccessful_notification).each do |method|
-    define_method method do
-      nil
+  def winners_reminder_to_submit(award_year)
```
Comment (matthewford) [inline app/services/notifiers/email_notification_service.rb]: Assignment Branch Condition size for winners_reminder_to_submit is too high. [16.06/15]

File: app/services/notifiers/email_notification_service.rb [hunk 7]
```diff
@@ -18,34 +18,73 @@ def run
...
+    award_year.form_answers.winners.each do |form_answer|
+      if form_answer.promotion?
+        Notifiers::Winners::PromotionBuckinghamPalaceInvite.perform_async(document["nominee_email"],                                           form_answer.id)
```
Comment (matthewford) [inline app/services/notifiers/email_notification_service.rb]: Line is too long. [158/100]<br>Unnecessary spacing detected.

File: app/services/notifiers/email_notification_service.rb [hunk 8]
```diff
@@ -18,34 +18,73 @@ def run
...
+      else
+        head_of_business = "#{document["head_of_business_first_name"]} #{document["head_of_business_last_name"]}"
```
Comment (matthewford) [inline app/services/notifiers/email_notification_service.rb]: Line is too long. [113/100]<br>Prefer single-quoted strings inside interpolations.

File: app/services/notifiers/email_notification_service.rb [hunk 9]
```diff
@@ -18,34 +18,73 @@ def run
...
+        Notifiers::Winners::BuckinghamPalaceInvite.perform_async(document["head_email"],
+                                                                 head_of_business,
+                                                                 form_answer.id)
+      end
+    end
+  end
+
+  def ep_reminder_support_letters(award_year)
+    award_year.form_answers.where(award_type: "promotion").includes(:support_letters).each do |form_answer|
```
Comment (matthewford) [inline app/services/notifiers/email_notification_service.rb]: Line is too long. [107/100]

File: app/services/notifiers/email_notification_service.rb [hunk 10]
```diff
@@ -54,11 +93,12 @@ def winners_notification(award_year)
       document = form_answer.document
 
       if form_answer.promotion?
-        Notifiers::Winners::PromotionBuckinghamPalaceInvite.perform_async(document["nominee_email"],
-                                                                          form_answer)
+        Notifiers::Winners::PromotionBuckinghamPalaceInvite.perform_async(document["nominee_email"],                                           form_answer.id)
```
Comment (matthewford) [inline app/services/notifiers/email_notification_service.rb]: Line is too long. [158/100]<br>Unnecessary spacing detected.

File: app/services/notifiers/email_notification_service.rb [hunk 11]
```diff
@@ -54,11 +93,12 @@ def winners_notification(award_year)
...
       else
+        head_of_business = "#{document["head_of_business_first_name"]} #{document["head_of_business_last_name"]}"
```
Comment (matthewford) [inline app/services/notifiers/email_notification_service.rb]: Line is too long. [113/100]<br>Prefer single-quoted strings inside interpolations.

File: app/services/notifiers/winners/buckingham_palace_invite.rb [hunk 12]
```diff
@@ -3,8 +3,8 @@ class Notifiers::Winners::BuckinghamPalaceInvite
 
   shoryuken_options queue: "#{Rails.env}_default", auto_delete: true
 
-  def perform(sqs_msg, email, form_answer_id)
+  def perform(sqs_msg, email, name, form_answer_id)
```
Comment (matthewford) [inline app/services/notifiers/winners/buckingham_palace_invite.rb]: Unused method argument - `sqs_msg`. If it's necessary, use `_` or `_sqs_msg` as an argument name to indicate that it won't be used.

File: spec/mailers/users/audit_certificate_request_mailer_spec.rb [hunk 13]
```diff
@@ -25,9 +31,9 @@
     end
 
     it "renders the body" do
-      expect(mail.body.encoded).to match(user.decorate.full_name)
-      expect(mail.body.encoded).to match(award_title)
-      expect(mail.body.encoded).to have_link("Review Audit Certificate", href: users_form_answer_audit_certificate_url(form_answer))
+      expect(mail.html_part.decoded).to match(user.decorate.full_name)
+      expect(mail.html_part.decoded).to have_link("upload your completed Audit Certificate here.", href: users_form_answer_audit_certificate_url(form_answer))
```
Comment (matthewford) [inline spec/mailers/users/audit_certificate_request_mailer_spec.rb]: Line is too long. [158/100]

File: spec/mailers/users/buckingham_palace_invite_mailer_spec.rb [hunk 14]
```diff
@@ -12,8 +20,9 @@
     end
 
     it "renders the body" do
-      expect(mail.body.encoded).to have_link("attendee details",
-                                             href: edit_palace_invite_url(id: palace_invite.token))
+      expect(mail.html_part.decoded).to have_link("the Winner's Dashboard.",
+                                             href: dashboard_url)
```
Comment (matthewford) [inline spec/mailers/users/buckingham_palace_invite_mailer_spec.rb]: Align the parameters of a method call if they span more than one line.

File: spec/mailers/users/notify_non_shortlisted_mailer_spec.rb [hunk 15]
```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+describe Users::NotifyNonShortlistedMailer do
+  let!(:user) { create :user }
+  let!(:form_answer) do
+    create :form_answer, :submitted, :innovation, user: user
+  end
+
+  let(:award_title) { form_answer.decorate.award_application_title }
+  let(:subject) {
```
Comment (matthewford) [inline spec/mailers/users/notify_non_shortlisted_mailer_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/mailers/users/notify_non_shortlisted_mailer_spec.rb [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    end
+
+    it "renders the body" do
+      expect(mail.html_part.decoded).to match(user.decorate.full_name)
+    end
+  end
+
+  describe "#ep_notify" do
+    let(:form_answer) { create :form_answer, :promotion, :submitted, user: user }
+    let(:subject) {
```
Comment (matthewford) [inline spec/mailers/users/notify_non_shortlisted_mailer_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/mailers/users/notify_shortlisted_mailer_spec.rb [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+
+    it "renders the headers" do
+      expect(mail.subject).to eq(subject)
+      expect(mail.to).to eq([user.email])
+      expect(mail.from).to eq(["no-reply@queens-awards-enterprise.service.gov.uk"])
+    end
+
+    it "renders the body" do
+      expect(mail.html_part.decoded).to match(user.decorate.full_name)
+      expect(mail.html_part.decoded).to have_link("upload here.", href: users_form_answer_audit_certificate_url(form_answer))
```
Comment (matthewford) [inline spec/mailers/users/notify_shortlisted_mailer_spec.rb]: Line is too long. [125/100]

File: spec/mailers/users/notify_shortlisted_mailer_spec.rb [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+  let!(:settings) { create :settings, :submission_deadlines }
+
+  let!(:deadline) do
+    deadline = Settings.current.deadlines.where(kind: "audit_certificates").first
+    deadline.update(trigger_at: Date.current)
+    deadline.trigger_at.strftime("%d/%m/%Y")
+  end
+
+  let(:award_title) { form_answer.decorate.award_application_title }
+  let(:subject) {
```
Comment (matthewford) [inline spec/mailers/users/notify_shortlisted_mailer_spec.rb]: Avoid using {...} for multi-line blocks.

File: spec/mailers/users/promotion_letters_of_support_reminder_spec.rb [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+
+  let!(:settings) { create :settings, :submission_deadlines }
+
+  let!(:deadline) do
+    Settings.current_submission_deadline.trigger_at.strftime("%d/%m/%Y")
+  end
+
+  before do
+    doc = form_answer.document
+    form_answer.document = doc.merge(nominee_info_first_name: "Jovan", nominee_info_last_name: "Savovich")
```
Comment (matthewford) [inline spec/mailers/users/promotion_letters_of_support_reminder_spec.rb]: Line is too long. [106/100]

File: spec/mailers/users/shortlisted_unsuccessful_feedback_mailer_spec.rb [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+  describe "#notify" do
+    let(:mail) { Users::ShortlistedUnsuccessfulFeedbackMailer.notify(form_answer.id) }
+
+    it "renders the headers" do
+      expect(mail.to).to eq([user.email])
+      expect(mail.from).to eq(["no-reply@queens-awards-enterprise.service.gov.uk"])
+    end
+
+    it "renders the body" do
+      expect(mail.html_part.decoded).to have_link("You can view your application feedback here.", href: "#")
```
Comment (matthewford) [inline spec/mailers/users/shortlisted_unsuccessful_feedback_mailer_spec.rb]: Line is too long. [108/100]

File: app/renderers/mail_renderer.rb [hunk 21]
```diff
@@ -9,20 +9,49 @@ def default_url_options
     end
   end
 
-  # this will be removed after all methods are implemented
-  %w(reminder_to_submit ep_reminder_support_letters winners_notification winners_reminder_to_submit unsuccessful_notification).each do |method|
-    define_method method do
-      "<b>TODO</b>".html_safe
-    end
+  def winners_reminder_to_submit
```
Comment (RuslanHamidullin) [inline app/renderers/mail_renderer.rb]:
```
+  def winners_reminder_to_submit
+    nil
+  end
```
Probably need to add "#TODO" prefix if it's not implemned

**Comment by RuslanHamidullin** [comment]
Looks good :+1: Waiting fixed spec

File: app/models/form_answer.rb [hunk 22]
```diff
@@ -161,6 +163,12 @@ def document
     super || {}
   end
 
+  def head_of_business
+    head_of_business = document["head_of_business_first_name"].to_s
+    head_of_business += " "
+    head_of_business += document["head_of_business_last_name"].to_s
```
Comment (matthewford) [inline app/models/form_answer.rb]: Useless assignment to variable - `head_of_business`. Use just operator `+`.

File: app/models/form_answer.rb [hunk 23]
```diff
@@ -99,17 +99,19 @@ def secondary
...
-    scope :shortlisted_with_no_certificate, -> { where("1 = 0") }
-    scope :winners, -> { where("1 = 0") }
-    scope :unsuccessful, -> { where(state: %w(not_recommended reserved)) }
+    scope :shortlisted, -> { where(state: %w(reserved recommended)) }
+    scope :not_shortlisted, -> { where(state: "not_recommended") }
+    scope :winners, -> { where(state: %(recomended awarded)) }
+    scope :unsuccessful, -> { where(state: %w(not_recommended not_awarded reserved)) }
     scope :submitted, -> { where(submitted: true) }
+    scope :business, -> { where(award_type: %w(trade innovation development)) }
+    scope :promotion, -> { where(award_type: "promotion") }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: spec/services/notifiers/email_notification_service_spec.rb [hunk 24]
```diff
@@ -154,4 +152,54 @@
...
+  context "ep_reminder_support_letters" do
+    let(:kind) { "ep_reminder_support_letters" }
+
+    let(:form_answer) do
+      create(:form_answer, :promotion)
+    end
+
+    it "triggers current notification" do
+      mailer = double(deliver_later!: true)
+      expect(Users::PromotionLettersOfSupportReminderMailer).to receive(:notify).with(form_answer.id) { mailer }
```
Comment (matthewford) [inline spec/services/notifiers/email_notification_service_spec.rb]: Line is too long. [112/100]

File: spec/services/notifiers/email_notification_service_spec.rb [hunk 25]
```diff
@@ -154,4 +152,54 @@
...
+
+  context "unsuccessful_notification" do
+    let(:kind) { "unsuccessful_notification" }
+
+    let(:form_answer) { create(:form_answer, :trade, state: "not_awarded") }
+    let!(:certificate) { create(:audit_certificate,form_answer: form_answer) }
+
+    it "triggers current notification" do
+      mailer = double(deliver_later!: true)
+      expect(Users::ShortlistedUnsuccessfulFeedbackMailer).to receive(:notify).with(form_answer.id) { mailer }
```
Comment (matthewford) [inline spec/services/notifiers/email_notification_service_spec.rb]: Line is too long. [110/100]

File: spec/services/notifiers/email_notification_service_spec.rb [hunk 26]
```diff
@@ -154,4 +152,54 @@
...
+
+      expect(current_notification.reload).to be_sent
+    end
+  end
+
+  context "unsuccessful_notification" do
+    let(:kind) { "unsuccessful_notification" }
+
+    let(:form_answer) { create(:form_answer, :trade, state: "not_awarded") }
+    let!(:certificate) { create(:audit_certificate,form_answer: form_answer) }
```
Comment (matthewford) [inline spec/services/notifiers/email_notification_service_spec.rb]: Space missing after comma.

**Comment by RuslanHamidullin** [comment]
Good stuff :+1:

**Pull request merged**